### ISSUE TITLE
it failes to load encrypted (S/MIME) msg file

### ISSUE THREAD

**Issue by kenjiuno** [Sicos1977/MSGReader#415]
**Describe the bug**
It throws an exception when I try to load the **encrypted** msg file by using `new MsgReader.Outlook.Storage.Message(...)`.

**To Reproduce**
As one repro sample, I have confirmed this behavior on Windows 10 with C# .NET 8.0 code:

ConsoleApp116.csproj

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="MsgReader" Version="5.6.4" />
  </ItemGroup>
</Project>

```

Program.cs

```cs
using System.Text;

namespace ConsoleApp116
{
    internal class Program
    {
        static void Main(string[] args)
        {
            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);

            using (var msg = new MsgReader.Outlook.Storage.Message(@"Message from smime1 to smime2 (encrypted).msg"))
            {
                var from = msg.Sender;
                var sentOn = msg.SentOn;
                var recipientsTo = msg.GetEmailRecipients(MsgReader.Outlook.RecipientType.To, false, false);
                var recipientsCc = msg.GetEmailRecipients(MsgReader.Outlook.RecipientType.Cc, false, false);
                var subject = msg.Subject;
                var htmlBody = msg.BodyHtml;
                // etc...
            }
        }
    }
}
```

**Expected behavior**

It can be read without exception thrown if there is an applicable situation including having a corresponding private key.

**Screenshots**

![2024-08-19_09h52_11](https://github.com/user-attachments/assets/3718c3a3-c380-45e8-b589-97d62af1659a)

```
Unhandled exception. System.Security.Cryptography.CryptographicException: Invalid cryptographic message type.
   at System.Security.Cryptography.Pkcs.SignedCms.<Decode>g__CopyContent|26_0(ReadOnlySpan`1 encodedMessage)
   at System.Security.Cryptography.Pkcs.SignedCms.Decode(ReadOnlySpan`1 encodedMessage)
   at System.Security.Cryptography.Pkcs.SignedCms.Decode(Byte[] encodedMessage)
   at MsgReader.Outlook.Storage.Message.ProcessSignedContent(Byte[] data)
   at MsgReader.Outlook.Storage.Message.LoadEncryptedAndPossibleSignedMessage(CFStorage storage)
   at MsgReader.Outlook.Storage.Message.LoadStorage(CFStorage storage)
   at MsgReader.Outlook.Storage..ctor(String storageFilePath, FileAccess fileAccess)
   at MsgReader.Outlook.Storage.Message..ctor(String msgfile, FileAccess fileAccess)
   at ConsoleApp116.Program.Main(String[] args) in H:\Proj\ConsoleApp116\ConsoleApp116\Program.cs:line 11

H:\Proj\ConsoleApp116\ConsoleApp116\bin\Debug\net8.0\ConsoleApp116.exe (プロセス 32748) は、コード -532462766 で終了しました。
このウィンドウを閉じるには、任意のキーを押してください...

```

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro (Japanese)
 - Browser: Google Chrome
 - Version: 127.0.6533.100

**Additional context**

The other day, in my GitHub repository, I received a [question](https://github.com/HiraokaHyperTools/msgreader/issues/49) about how to deal with `smime.p7m` file.
At that time, I didn't know how to decode it. So I studied about it.
And then I created a complete set of S/MIME sample data at my [repository](https://github.com/HiraokaHyperTools/smime_mail_samples) for future research.
Because this MSGReader project seems to contain decryption code, I have tested it.

**Comment by Sicos1977** [comment]
I dont understand what the problem is and what I should do to fix it?

**Comment by kenjiuno** [comment]
Hi.

This is a sample msg file encrypted with S/MIME

https://github.com/HiraokaHyperTools/smime_mail_samples/blob/master/received-at-smime2/Message%20from%20smime1%20to%20smime2%20(encrypted).msg

I can open this by my Office Outlook

![2025-01-09_16h48_44](https://github.com/user-attachments/assets/e2fdf881-e3a0-4ea8-9cb2-1c7c65a6b457)

This msg is encrypted with a self-signed certificate.

If you need certs and instructions about how to, I have prepared all of them on my repository.

https://github.com/HiraokaHyperTools/smime_mail_samples/

**Comment by kenjiuno** [comment]
This story I'm talking about is unrelated to you, but I'll write here because I thought I should.

I maintain the forked version of [msgreader](https://www.npmjs.com/package/@kenjiuno/msgreader) (JavaScript version) available to JavaScript developers.

Although my company uses the JavaScript version of msgreader, the original authors do not maintain it anymore.
So I maintain the forked version instead.

Often I receive questions that only an expert would understand.
The following issue mentions about S/MIME thing that I didn't know then.

https://github.com/HiraokaHyperTools/msgreader/issues/49

Because I'm interested in technical reasons (S/MIME can be processed upon JavaScript environment?), I investigated it.

I have purchased DigiCert Class 1 S/MIME 1 year (3,500 JPY w/ tax) for this purpose.
Thanks to having genuine certification, I could observe precisely how desktop mailers (Thunderbird and Outlook) behave about S/MIME signed/encrypted messages.

And I wanted to leave the info I have acquired for future use.
Because I cannot share the actual certification, I have crafted self-signed certifications.
And I have acquired the S/MIME processed mail messages by using them.
